release/ray_release/test_automation/state_machine.py

```diff
@@ -102,6 +102,8 @@ def _move_hook(self, from_state: TestState, to_state: TestState) -> None:
         elif change == (TestState.JAILED, TestState.PASSING):
             self._close_github_issue()
             self.test.pop(Test.KEY_BISECT_BUILD_NUMBER, None)
+        elif change == (TestState.JAILED, TestState.JAILED):
```

Comment:
this is a "no change"? does all no change also trigger these hooks?

can this hook be separated? like in a form of "state hook" rather than a "change hook"?

Reply:
Yeah this is a no change, but this hook is triggered only when the test itself moves between the same state. Not sure what you have in mind for state hook, but the state has to move for this hook to be triggered. Consider this scenario:

test -> JAILED (then test stops running anywhere so no more new test result, no more transition)

test owner claimed test is fixed, close the associated github issue

test now runs again, just to realize that it still fails, so it moves from JAILED to JAILED itself. Then this hook is triggered to re-open the github issue again

Reply:
synced offline. idea is that we will add a "state_hook" and `keep_github_issue_open` when the `to_state` is `JAIL`, no matter what the `from_state` is.